elevate-cpanel

```diff
@@ -3534,6 +3534,106 @@ EOS
+            foreach my $key ( keys %opt_args ) {
+                $opts{$key} = $opt_args{$key};
+            }
+        }
+
+        unshift @args, GRUBBY_PATH;
+
+        return $opts{die_on_error} ? $self->ssystem_and_die(@args) : $self->ssystem( \%opts, @args );
+    }
+
+    sub _default_kernel ($self) {
+        return $self->_call_grubby( { should_capture_output => 1, should_hide_output => 1 }, '--default-kernel' )->{'stdout'}->[0] // '';
+    }
+
+    sub _persistent_id {
+        my $id = Elevate::StageFile::read_stage_file( 'bootloader_random_tag', '' );
+        return $id if $id;
+
+        $id = int( rand(100000) );
+        Elevate::StageFile::update_stage_file( { 'bootloader_random_tag', $id } );
+        return $id;
+    }
+
+    sub mark_cmdline ($self) {
+        my $arg = "elevate-" . _persistent_id;
+        INFO("Marking default boot entry with additional parameter \"$arg\".");
+
+        my $kernel_path = $self->_default_kernel;
+        $self->_call_grubby( { die_on_error => 1 }, "--update-kernel=$kernel_path", "--args=$arg" );
+
+        return;
+    }
+
+    sub verify_cmdline ($self) {
+        my $arg = "elevate-" . _persistent_id;
+        INFO("Checking for \"$arg\" in booted kernel's command line...");
+
+        my $kernel_cmdline = eval { File::Slurper::read_binary(CMDLINE_PATH) };
+        LOGDIE( CMDLINE_PATH . ' is unexpectedly empty; leapp may not support this system' ) unless $kernel_cmdline;
+        DEBUG( CMDLINE_PATH . " contains: $kernel_cmdline" );
+
+        my $detected = scalar grep { $_ eq $arg } split( ' ', $kernel_cmdline );
+        if ($detected) {
+            INFO("Parameter detected; restoring entry to original state.");
+        }
+        else {
+            ERROR("Parameter not detected. Attempt to upgrade is being aborted.");
+        }
+
+        my $kernel_path = $self->_default_kernel;
+        my $result      = $self->_call_grubby( "--update-kernel=$kernel_path", "--remove-args=$arg" );
+        WARN("Unable to restore original command line. This should not cause problems but is unusual.") if $result != 0;
+
+        if ( !$detected ) {
+
+            my $stage              = Elevate::Stages::get_stage();
+            my $pretty_distro_name = $self->cpev->upgrade_to_pretty_name();
+            my $msg                = <<"EOS";
+The elevation process failed during stage $stage.
+
+Specifically, it was not possible to prove that the system has control over its own boot process using the utilities the operating system provides.
+
+For this reason, the elevation process has been terminated before making any unreversable changes.
+
+You can check the error log by running:
+
+    $0
+
+If it is possible to restore the ability for the system to manipulate the boot process, you can restart the elevation process anew:
+
+    $0 --start
+
+EOS
+            Elevate::Notify::send_notification( qq[Failed to update to $pretty_distro_name] => $msg );
+
+            $self->cpev->do_cleanup(1);
```

Comment:
The issue is that invoking `remove` necessarily invokes `stop`: https://github.com/cpanel/elevate/blob/d312cdc7b88ee46c3ec8965698847e55e8d65039/lib/Elevate/SystemctlService.pm#L76

Since we are invoking `do_cleanup` in the service context, allowing `stop` to run is likely to result in terminating the script before it finishes its cleanup. By contrast, `disable` also removes the service file, but it does not stop the service by default. We can thus let the service terminate on its own by reaching `exit`.

Yes, this is totally brain-dead, but re-doing the service handling code is more scope creep than I needed.